### PR TITLE
Validate wsi in BUFR pipeline

### PR DESCRIPTION
**Pull request by webb-ben**
re: https://github.com/wmo-im/wis2box/issues/208
@tomkralidis @david-i-berry @maaikelimper would love if you could test this branch against more data. The two additional test datasets consist of 43 subsets of messages. The only failure is on wsi: ``0-20000-0-60577``. 
- Validate WSI & cache using stations_list.csv
- Add geometry to BUFR if missing
- Add more stations to station_list.csv (to speed up tests)
- Add more test data and update test workflow to reflect additional messages published.

### CODE REVIEW

File: wis2box/metadata/station.py [hunk 1]
```diff
@@ -178,6 +173,67 @@ def publish_station_collection() -> None:
     return
 
 
+def validate_wsi(wsi):
```
Comment (tomkralidis) [inline wis2box/metadata/station.py:176]: Add function annotation

File: wis2box/metadata/station.py [hunk 2]
```diff
@@ -178,6 +173,67 @@ def publish_station_collection() -> None:
...
+    """Validates and caches WSI from WMO OSCAR/Surface"""
```
Comment (tomkralidis) [inline wis2box/metadata/station.py:177]: Update to proper docstring

File: wis2box/metadata/station.py [hunk 3]
```diff
@@ -178,6 +173,67 @@ def publish_station_collection() -> None:
...
+    stations = DATADIR / 'metadata' / 'station' / 'station_list.csv'
+
+    LOGGER.info(f'Validating WIGOS Station Identifier: {wsi}')
+    with stations.open('r', newline='') as fh:
```
Comment (tomkralidis) [inline wis2box/metadata/station.py:181]: we can omit `r` given it is the default.

File: wis2box/metadata/station.py [hunk 4]
```diff
@@ -178,6 +173,67 @@ def publish_station_collection() -> None:
...
+
+    filename = DATADIR / 'metadata' / 'station' / f'{wsi}.json'
+    LOGGER.debug(f'Caching station report to {filename}')
+    with filename.open('w') as fh:
+        json.dump(station_report, fh)
+
+    return True
+
+
+def get_geometry(wsi):
```
Comment (tomkralidis) [inline wis2box/metadata/station.py:208]: Add function annotation

File: wis2box/metadata/station.py [hunk 5]
```diff
@@ -178,6 +173,67 @@ def publish_station_collection() -> None:
...
+    """Validates and caches station geometry from WMO OSCAR/Surface"""
```
Comment (tomkralidis) [inline wis2box/metadata/station.py:209]: Update with proper docstring

**Review by tomkralidis — APPROVED** [review]

File: wis2box/metadata/station.py [hunk 6]
```diff
@@ -179,6 +184,87 @@ def publish_station_collection() -> None:
...
+    :returns: `dict`, of station geometr
+    """
+
+    filename = STATION_METADATA / f'{wsi}.json'
+
+    if filename.exists():
+        with filename.open() as fh:
+            station_report = json.load(fh)
+    else:
+        cache_station(wsi)
```
Comment (david-i-berry) [inline wis2box/metadata/station.py:245]: Check cache_station() return value and if False return None

**Review by david-i-berry — APPROVED** [review]

**Pull request merged**